### DISCUSSION TITLE
Get all links for a domain

### DISCUSSION THREAD

**Discussion by gagegogan** [shlinkio/shlink#2178] [Help wanted] [answered]
### Shlink version

Latest

### PHP version

Any

### How do you serve Shlink

Self-hosted nginx

### Database engine

MySQL

### Database version

Doesn’t matter

### Summary

I was reading the documentation and I didn’t see a filter to get all links for a given domain. Is that currently possible?

**Comment by acelaya** [comment, marked as answer]
Technically speaking, you can get them by passing the domain name to the `searchTerm` query parameter (the documentation here https://api-spec.shlink.io/#/Short%20URLs/listShortUrls is outdated and doesn't mention it. I need to fix that), although it's possible you get some extra results if using that domain as tag, custom slug, title, etc. (unlikely, but not impossible).

A couple years ago I considered adding an [explicit filter for domain](https://github.com/shlinkio/shlink/issues/1520), but ended up closing it as it was more or less possible to do as described above.

I may reopen it, as it seems like a useful thing to have regardless.
